### DISCUSSION TITLE
Exports should have an option to export the current columns

### DISCUSSION THREAD

**Discussion by engram-design** [craftcms/cms#9758] [General]
We have a few clients that want to export items from entries, orders and some other [custom elements](https://github.com/verbb/events), but they're getting confused by the output not being what they expect.

From a client point of view, they're expecting the export to contain **just** the columns they're currently viewing in their control panel. Now granted, the default exports contain columns for every custom field available on the system and every attribute on the element, and some might think that useful - more is better than less. And the client can certainly go through and pick-and-choose the columns to keep, but that's pretty tedious on a system with lots of custom fields.

I'll mention that of course we can create a custom bit of code to add a new type of export to handle this scenario (particularly for third-party plugins), but feel like maybe it should be the default export behaviour? Even if it's a breaking change?

Related, but is there much point - other than performance - for Raw and Expanded? There's columns in the default exports like `fieldLayoutId`, `newParentId`, `deletedWithEntryType`, `tempId`, `draftId`, `contentId`, etc - all of which are pretty unnecessary. These two exports seem pretty identical otherwise.

What I might propose is to have 2 exports, both of which are different to the current.

1. Default - exports whatever the current view of columns the user has picked and can see in the control panel. This lets them "prepare" a report of their chosen data. Number of rows unchanged of course.
2. Expanded - All columns available to be added (through the cog icon) should be listed. This handles the case of when exporting across multiple sections/sources by using only the source columns available, it doesn't contain columns that are internal and columns can have proper headings (field_someField = "Some Field" and sectionId = "Section"). It also handles columns added by other plugins (Votes), or ones added in modules.

So maybe something to consider or discuss?

**Comment by wmdhosting** [comment]
agree with this one..   example. 
I have section with 5 fields..    and there is 143 rows exported.  JSON
But no category.. for example so exports only  4 fields 


![Snag_100d69b4](https://user-images.githubusercontent.com/76176291/202254657-9857052a-4d7d-466f-820f-47390f0e251b.png)

`     "id": "109687",
        "canonicalId": null,
        "fieldLayoutId": "274",
        "uid": "d5e99365-f323-44b7-9605-b176876e187c",
        "enabled": "1",
        "archived": "0",
        "dateLastMerged": null,
        "dateCreated": "2022-11-05 06:39:02",
        "dateUpdated": "2022-11-16 17:19:53",
        "siteSettingsId": "276760",
        "slug": "repair-3",
        "siteId": "1",
        "uri": "domena/repair-3",
        "enabledForSite": "1",
        "sectionId": "33",
        "typeId": "71",
        "authorId": "1",
        "postDate": "2022-11-05 06:39:00",
        "expiryDate": null,
        "contentId": "77653",
        "title": ".repair",
        "field_articleEditor_rvxyioml": null,
        "field_autoplay": null,
        "field_backgroundColor": null,
        "field_bigCursor": null,
        "field_bigCursorClass": null,
        "field_bodyScripts": null,
        "field_typeCard": null,
        "field_typeCardCat_ypjredbc": null,
        "field_buttonPosition_rtxiljbw": null,
        "field_buttonsBackgroundColor_lvfawxws": null,
        "field_category": null,
        "field_categories": null,
        "field_categoryLevel_joqhdlis": null,
        "field_codeEditor": null,
        "field_size": null,
        "field_sadrzaj": null,
        "field_contrastMenu": null,
        "field_contrastMenuClass": null,
        "field_controls": null,
        "field_countEntry": null,
        "field_class": null,
        "field_cursorDefault": null,
        "field_cursorDefaultClass": null,
        "field_cursorMenu": null,
        "field_customSlug_ocrxzvir": null,
        "field_partsOfPath_asurtgco": null,
        "field_darkContrast": null,
        "field_darkContrastClass": null,
        "field_showDate_dcepohxk": null,
        "field_dateFormat_dpvivbao": null,
        "field_datePosition_mpzhvonr": null,
        "field_defaultFont": null,
        "field_defaultFontClass": null,
        "field_defaultFontSize": null,
        "field_defaultFontSizeClass": null,
        "field_defaultStyle": null,
        "field_defaultStyleClass": null,
        "field_description": null,
        "field_descriptionCategory": null,
        "field_descriptionsACF": null,
        "field_domainPrices": null,
        "field_dropdownACF": null,
        "field_dyslexiaFont": null,
        "field_dyslexiaFontClass": null,
        "field_fontMenu": null,
        "field_fontSizeBig": null,
        "field_fontSizeBigClass": null,
        "field_fontSizeMenu": null,
        "field_fontSizeSmall": null,
        "field_fontSizeSmallClass": null,
        "field_footerScripts": null,
        "field_headLibs": null,
        "field_header": null,
        "field_customStyles_raqvstbu": null,
        "field_headerScripts": null,
        "field_icon": null,
        "field_iconClass": null,
        "field_iconForListItem": null,
        "field_instructionForAcfFields": null,
        "field_specifications": null,
        "field_latestFromTwitter": null,
        "field_lightContrast": null,
        "field_lightContrastClass": null,
        "field_linkOnFacebook": null,
        "field_liveMode": null,
        "field_localeAccessibility_kdrpeuqx": null,
        "field_mapType": null,
        "field_thumbType_wuollxnx": null,
        "field_menuList_ryvybise": null,
        "field_navigationType_cyojeiti": null,
        "field_pageBuilderInVizy": null,
        "field_photosFromInstagram": null,
        "field_plainText2ACF": null,
        "field_plainTextACF": null,
        "field_positionPopUp": null,
        "field_popupAnimation_rmnpjvul": null,
        "field_priceInEur": "59.00",
        "field_priceInHrk": "385.00",
        "field_redACF": null,
        "field_redactorACF": null,
        "field_reset": null,
        "field_resetSettingsClass": null,
        "field_overlay": null,
        "field_sectionContent": null,
        "field_sectionForSearch": null,
        "field_sectionOfEntries": null,
        "field_seo": null,
        "field_seoDescription": null,
        "field_show_nlkvebqe": null,
        "field_showAfterVisits": null,
        "field_showAlways": null,
        "field_notShowNumberOfEntries": null,
        "field_device": null,
        "field_selectedCategories_stvkmxfq": null,
        "field_popUpType": null,
        "field_showSubcat_totugebm": null,
        "field_typeImagesWithLink": null,
        "field_sidebarAlign_rxvwofns": null,
        "field_siftaCONNECTACF": null,
        "field_skinCss": null,
        "field_slideButtonPosition_kqxiuwxx": null,
        "field_rightSide": null,
        "field_socialMedia": null,
        "field_styleSwitcherOnOff": null,
        "field_styleUnderline": null,
        "field_styleUnderlineClass": null,
        "field_stylesMenu": null,
        "field_switcherHeader": null,
        "field_tagGroup": null,
        "field_separator_cbkguhdq": null,
        "field_templateAccessibility_iqqrmewj": null,
        "field_temporaryList": null,
        "field_text": null,
        "field_textColor_xagxlaxj": null,
        "field_tls": ".repair",
        "field_useIconLikeLink": null,
        "field_vizy": null,
        "draftId": null,
        "draftCreatorId": null,
        "isProvisionalDraft": null,
        "draftName": null,
        "draftNotes": nul`

**Comment by thisisjamessmith** [comment]
Just wondering if there are any plans to look at this for Craft 5+? Appreciate there are probably some complex technical barriers here.

**Reply by brandonkelly** [reply]
We want to completely overhaul exporting (in conjunction with importing) at some point soon, so we can consider this when we get there.
